lib/shared/pe_status_check.rb

```diff
@@ -66,33 +71,43 @@ def self.service_file_exist?(configfile)
     File.exist?("#{configdir}/#{configfile}")
   end
 
-  # Queries the passed port and endpoint for the status API
+  # Module method to make a GET request to an api specified by path and port params
   # @return [Hash] Response body of the API call
-  # param port [Integer] The status API port to query
-  # @param endpoint [String] The status API endpoint to query
-  def self.status_check(port, endpoint)
-    require 'json'
-
-    host     = Puppet[:certname]
-    client   = Puppet.runtime[:http]
-    response = client.get(URI(Puppet::Util.uri_encode("https://#{host}:#{port}/status/v1/services/#{endpoint}")))
-    status   = JSON.parse(response.body)
-    status
-  rescue Puppet::HTTP::ResponseError => e
-    Facter.debug("fact 'self_service' - HTTP: #{e.response.code} #{e.response.reason}")
-  rescue Puppet::HTTP::ConnectionError => e
-    Facter.debug("fact 'self_service' - Connection error: #{e.message}")
-  rescue Puppet::SSL::SSLError => e
-    Facter.debug("fact 'self_service' - SSL error: #{e.message}")
-  rescue Puppet::HTTP::HTTPError => e
-    Facter.debug("fact 'self_service' - General HTTP error: #{e.message}")
-  rescue JSON::ParserError => e
-    Facter.debug("fact 'self_service' - Could not parse body for JSON: #{e}")
+  # @param path [String] The API path to query.  Should include a '/' prefix and query parameters
+  # @param port [Integer] The port to use
+  # @param host [String] The FQDN to use in making the connection.  Defaults to the Puppet certname
+  def http_get(path, port, host = Puppet[:certname])
+    # Use an instance variable to only create an SSLContext once
+    @ssl_context ||= Puppet::SSL::SSLContext.new(
+      cacerts: Puppet[:localcacert],
+      private_key: OpenSSL::PKey::RSA.new(File.read(Puppet[:hostprivkey])),
+      client_cert: OpenSSL::X509::Certificate.new(File.open(Puppet[:hostcert])),
+    )
+
+    client = Net::HTTP.new(host, port)
+    # The main reason to use this approach is to set open and read timeouts to a small value
+    # Puppet's HTTP client does not allow access to these
+    client.open_timeout = 2
+    client.read_timeout = 2
+    client.use_ssl = true
+    client.verify_mode = OpenSSL::SSL::VERIFY_PEER
+    client.cert = @ssl_context.client_cert
+    client.key = @ssl_context.private_key
+    client.ca_file = @ssl_context.cacerts
+
+    response = client.request_get(Puppet::Util.uri_encode(path))
+    if response.is_a? Net::HTTPSuccess
+      JSON.parse(response.body)
+    else
+      false
+    end
+  rescue StandardError => e
+    Facter.warn("Error in fact 'self_service': #{e.backtrace}")
   end
 
   # Check if Primary node
   # @return [Boolean] true is primary node
-  def self.primary?
+  def primary?
```

Comment:
Hi @m0dular.  I've noticed that you've changed the class methods to instance ones.  I'm sure you've already done this, but have you made sure to create an instance of the PeStatusCheck when calling these methods now?

```
➜  ruby-usage-tutorials git:(main) ✗ rconsole
[1] pry(main)> class Bob
[1] pry(main)*   def self.class_method
[1] pry(main)*     "I'm a  class method"
[1] pry(main)*   end  
[1] pry(main)*   def instance_method
[1] pry(main)*     "I'm an instance method"
[1] pry(main)*   end  
[1] pry(main)* end  
=> :instance_method
[2] pry(main)> Bob.class_method
=> "I'm a  class method"
[3] pry(main)> Bob.instance_method
ArgumentError: wrong number of arguments (given 0, expected 1)
from (pry):10:in `instance_method'
[4] pry(main)> bob = Bob.new
=> #<Bob:0x00007fdca4287e58>
[5] pry(main)> bob.instance_method
=> "I'm an instance method"
[6] pry(main)> 
```